source/features/filter-comments-by-you.tsx

```diff
@@ -1,21 +1,38 @@
 import React from 'dom-chef';
 import select from 'select-dom';
 import features from '../libs/features';
-import {getRepoURL} from '../libs/utils';
+import {getUsername, getRepoURL} from '../libs/utils';
+import {isEnterprise} from '../libs/page-detect';
 
 const repoUrl = getRepoURL();
 
 function init(): void {
-	select('.subnav-search-context .SelectMenu-list a:last-child')!
-		.before(
-			<a
-				href={`/${repoUrl}/issues?q=is%3Aopen+commenter:@me`}
-				className="SelectMenu-item"
-				role="menuitem"
-			>
-				Everything commented by you
-			</a>
-		);
+	if (isEnterprise()) {
```

Comment:
There is a lot of duplication here. The key differences here are:
- the selector: this can be joined together
- the `<li>` wrapper: this isn't needed on 2.19
- The classname of `SelectMenu-Item`: this can be applied to both if needed and again, this markup is different on 2.19

This needs to be rewritten in a more generic way to apply to GH and differing versions of GHE.

Reply:
Yeah I thought about writing GE version retriever, and do it more generic then

Reply:
We don't want to go down the route of coding to specific versions of GHE. Code so it works for both and eventually GHE will catch up and that code can be removed.

Reply:
the question is, do we want to support old versions of GHE. Because some enterprises may keep a specific GHE version for a longer time although there are maybe like 5 newer versions available, so the catch up will not happen

Reply:
Supporting older is fine, but it needs to be done in a better way than using `isEnterprise`.

Reply:
but the classname `SelectMenu-Item` is unfortunatly not usable for both, `SelectMenu-Item` (new) vs `select-menu-item` (old)

i am trying to find a possibility

Reply:
You can pass two selectors to the select, this is better than two branching if statements with 99% of the code the same. 😀